framework/CMakeLists.txt

```diff
@@ -402,9 +402,9 @@ endif()
 if(${VKB_WARNINGS_AS_ERRORS})
     message(STATUS "Warnings as Errors Enabled")
     if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-        target_compile_options(${PROJECT_NAME} PRIVATE -Werror)
+        target_compile_options(${PROJECT_NAME} PRIVATE -Werror -Wall)
     elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
-        target_compile_options(${PROJECT_NAME} PRIVATE /W3 /WX)
+        target_compile_options(${PROJECT_NAME} PRIVATE /Wall /WX)
```

Comment:
Been a bit since last I checked on what compile flags MSVC supports, However, I thought Wall is clang/gcc only and msvc would use /W4

As there's going to be several warnings, would you like some help on the cleanup?  We can all use this branch after we get the plugins PR merged.

Reply:
MSVC does indeed support /Qall. I'm not sure if we actually want to use that. Details on this can be found here: https://docs.microsoft.com/en-us/cpp/preprocessor/compiler-warnings-that-are-off-by-default?view=msvc-160

If we enable this for MSVC we'll probably need to explicitly disable different warnings, esp. with VS2019.

Reply:
The trick we'll need to use is to specify that the compile options are private the the framework and samples so they don't also go into the dependencies.  That should cut down quite a bit.

Looking locally I'm seeing a fair number that can be auto resolved (clion has these tools... ;-)).  Mostly static analysis tools can get rid of very simple warnings.  However, I'm more concerned with the false positives.  We'll need to target very specific parts of the code with a macro that disables specific warnings as there will be several instances of unused variable that gets set in C++ but used in shaders.